lib/ably/auth.rb

```diff
@@ -510,11 +510,11 @@ def token_option
     end
 
     def authorize_when_necessary
-      if current_token_details && !current_token_details.expired?
+      if current_token_details && !current_token_details.expired?(from: current_time)
```

Comment:
Don't we need to double-check here that `@server_time_offset` is present to make sure that current_time is based on server time?

Reply:
I have checked local build. It always calls for `ensure_current_time_is_based_on_server_time` if `query_time` and requesting for a token.